storage/fcds/fcds.go

```diff
@@ -36,8 +38,9 @@ import (
 type Storer interface {
 	Get(addr chunk.Address) (ch chunk.Chunk, err error)
 	Has(addr chunk.Address) (yes bool, err error)
-	Put(ch chunk.Chunk) (err error)
+	Put(ch chunk.Chunk) (shard uint8, err error)
 	Delete(addr chunk.Address) (err error)
+	ShardSize() (slots []ShardInfo, err error)
```

Comment:
correct, but if we want to be able to test the package at the edges then we have to leak this information

Reply:
Yes, I am wondering if that could be done with another interface which would be used for this case. ShardSize method is easy to pull out, but Put method is the tricky one. I suppose that it would be too much complexity to implement an event on which shard the chunk is put, and put a subscription on the shard related interface.